File: senior/contests/Diff.py
def main():
    for i in range(5):
        strA = input("Enter the first string: ")
        strB = input("Enter the second string: ")

        # diff A to B
        arrA = strA.split(" ")
        copyB = strB
        commonStr1 = ""
        for a in arrA:
            if a in copyB:
                wordIndex = copyB.index(a)
                copyB = copyB[:wordIndex] + copyB[wordIndex + len(a):]
                commonStr1 += a + " "
        commonStr1 = commonStr1[:len(commonStr1) - 1]

        # diff B to A
        arrB = strB.split(" ")
        copyA = strA
        commonStr2 = ""
        for b in arrB:
            if b in copyA:
                wordIndex = copyA.index(b)
                copyA = copyA[:wordIndex] + copyA[wordIndex + len(b):]
                commonStr2 += b + " "
        commonStr2 = commonStr2[:len(commonStr2) - 1]

        # implement second algorithm
        finalStr = ""
        for c in commonStr1:
            if c != " ":
                if c in commonStr2:
                    finalStr += c
                    commonStr2 = commonStr2[commonStr2.index(c) + 1:]
        if len(finalStr) == 0:
            print("NONE")
        else:
            print(finalStr)

main()

'''
The quick brown fox did jump over a log
The brown rabbit quickly did outjump the fox

How many pickled peppers did Peter Piper pick
Peter Piper picked a peck of pickled peppers

The AllStar Contest is in Wayne Hills NJ
Hills are where there are contestants from NJ

Fuzzy Wuzzy was a bear Fuzzy Wuzzy had no hair
Fuzzy Wuzzy was not fuzzy was he

Super cali frigi listic expi alli docious
frigid call superman allies expired list
'''

File: senior/contests/Diff.java
import java.io.*;
import java.util.*;

public class Diff {
    public static void main(String[] args) {
        for(int i = 0; i < 5; i++) {
            Scanner input = new Scanner(System.in);
            System.out.print("Enter the first string: ");
            String strA = input.nextLine();
            System.out.print("Enter the second string: ");
            String strB = input.nextLine();

            String[] arrA = strA.split(" ");
            String copyB = strB;
            String commonStr1 = "";
            for(int j = 0; j < arrA.length; j++) {
                if(copyB.indexOf(arrA[j]) != -1) {
                    int wordIndex = copyB.indexOf(arrA[j]);
                    copyB = copyB.substring(0, wordIndex) + copyB.substring(wordIndex + arrA[j].length());
                    commonStr1 += arrA[j] + " ";
                }
            }
            commonStr1 = commonStr1.substring(0, commonStr1.length() - 1);

            String[] arrB = strB.split(" ");
            String copyA = strA;
            String commonStr2 = "";
            for(int j = 0; j < arrB.length; j++) {
                if(copyA.indexOf(arrB[j]) != -1) {
                    int wordIndex = copyA.indexOf(arrB[j]);
                    copyA = copyA.substring(0, wordIndex) + copyA.substring(wordIndex + arrB[j].length());
                    commonStr2 += arrB[j] + " ";
                }
            }
            commonStr2 = commonStr2.substring(0, commonStr2.length() - 1);

            String finalStr = "";
            String[] c = commonStr1.split("");
            for(int j = 1; j < c.length; j++) {
                if(!c[j].equals(" ")) {
                    if(commonStr2.indexOf(c[j]) != -1) {
                        finalStr += c[j];
                        commonStr2 = commonStr2.substring(commonStr2.indexOf(c[j]) + 1);
                    }
                }
            }
            if(finalStr.length() == 0) {
                System.out.println("NONE");
            } else {
                System.out.println(finalStr);
            }

        }
    }
}
